doc/dev/transforms.rst

```diff
@@ -100,6 +100,117 @@ classical instructions as well as side-effecting operations (e.g., print) can al
+
+Now we write the pass itself. Create a new file ``mlir/lib/Catalyst/Transforms/MyHelloWorldPass.cpp`` with the following content:
+
+.. code-block:: cpp
+
+    #define DEBUG_TYPE "myhelloworld"
+
+    #include "Catalyst/IR/CatalystDialect.h"
+    #include "mlir/Pass/Pass.h"
+    #include "llvm/Support/Debug.h"
+
+    using namespace llvm;
+    using namespace mlir;
+    using namespace catalyst;
+
+    namespace catalyst {
+    #define GEN_PASS_DEF_MYHELLOWORLDPASS
+    #include "Catalyst/Transforms/Passes.h.inc"
+
+    struct MyHelloWorldPass : public impl::MyHelloWorldPassBase<MyHelloWorldPass> {
+        using impl::MyHelloWorldPassBase<MyHelloWorldPass>::MyHelloWorldPassBase;
+
+        void runOnOperation() override { llvm::errs() << "Hello world!\n"; }
+    };
+
+    std::unique_ptr<Pass> createMyHelloWorldPass() { return std::make_unique<MyHelloWorldPass>(); }
+
+    } // namespace catalyst
+
+The function that determines what your pass actually does is the ``void runOnOperation()``. Here all the pass does is print out ``"Hello world!\n"``. 
+
+This new C++ file needs to be added to the ``mlir/lib/Catalyst/Transforms/CMakeLists.txt`` file (or the CMakeLists.txt of whichever directory that has your new pass file): 
+
+.. code-block::
+
+    file(GLOB SRC
+        ...
+        MyHelloWorldPass.cpp
+    )
+
+After writing the pass, we need to register it in a few places. In ``mlir/include/Catalyst/Transforms/Passes.h``, add the method 
+
+.. code-block:: cpp
+
+    namespace catalyst {
+        ...
+        std::unique_ptr<mlir::Pass> createMyHelloWorldPass();
+        ...
+    }
+
+And in ``mlir/lib/Catalyst/Transforms/RegisterAllPasses.cpp``, register the pass via 
+
+.. code-block:: cpp
+
+    void catalyst::registerAllCatalystPasses()
+    {
+        ...
+        mlir::registerPass(catalyst::createMyHelloWorldPass);
+        ...
+    }
+
+Now that we have written our shiny new pass, we can build it by going back to the top-level ``catalyst`` directory and 
+
+.. code-block::
+
+    make dialects
+
+The tool to run passes is built as ``mlir/build/bin/quantum-opt``.
+
+We can inspect by all the available passes by running ``quantum-opt --help``:
+
+.. code-block::
+
+    OVERVIEW: Quantum optimizer driver
+    ...
+    USAGE: quantum-opt [options] <input file>
+
+    OPTIONS:
+        ...
+        --my-hello-world                   -   An empty pass boilerplate that prints out hello world.
+
+Here the displayed ``--help`` message will be the ``summary`` we wrote in the tablegen file. The command line option to run our new pass is the template string in the def line in the tablegen file. 
+
+To run the pass, simply do 
+
+.. code-block::
+
+    quantum-opt -my-hello-world input.mlir
```

Comment:
```suggestion
    ./mlir/build/bin/quantum-opt -my-hello-world input.mlir
```


What is the `input.mlir` file expected to contain? I guess you can in principle just run the pass with an empty file? Then I get

```
% ./mlir/build/bin/quantum-opt --my-hello-world hackathon/empty.mlir
Hello world!
module {
}
```

Reply:
Yes, any input mlir file will do